src/testing/testpachd/main.go

```diff
@@ -0,0 +1,121 @@
+// Command testpachd starts up a local pachd.  It prints its address to stdout when it is ready and
+// then runs forever.  It will exit if it encounters an error, or if Control-C is pressed.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/config"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachd"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"go.uber.org/zap"
+)
+
+var (
+	logfile = flag.String("log", "", "if set, log to this file")
+	pachCtx = flag.String("context", "testpachd", "if set, setup the named pach context to connect to this server, and switch to it")
+)
+
+func main() {
+	flag.Parse()
+	done := log.InitBatchLogger(*logfile)
+	ctx, cancel := pctx.Interactive()
+	pd, cleanup, err := pachd.BuildTestPachd(ctx)
```

Comment:
The cleanup is done with defer now.  This will clean up under a few extra circumstances (can't write pach config, etc.), so that's good.

I did add defer/recover around pd.Run, but the RPCs don't actually run in that goroutine, so it has no real effect.  We need to add the panic interceptor for that, which we can do.